bot/cogs/monstersurvey.py

```diff
@@ -0,0 +1,161 @@
+                      'at a time.',
+                inline=False
+            )
+            default_embed.add_field(
+                name='.monster leaderboard',
+                value='Which monster has the most votes? This command will tell you.',
+                inline=False
+            )
+            return await ctx.send(embed=default_embed)
+
+    @monster_group.command(
+        name='vote'
+    )
+    async def monster_vote(self, ctx: Context, name: Optional[str] = None):
+        """Casts a vote for a particular monster, or displays a list of monsters that can be voted for
+        if one is not given."""
+        vote_embed = Embed(
+            name='Monster Voting',
+            color=0xFF6800
+        )
+        if name is None or name.lower() not in self.voter_registry.keys():
+            if name is not None:
+                vote_embed.description = f'You cannot vote for {name} because it\'s not in the running.'
+            vote_embed.add_field(
+                name='Use `.monster show {monster_name}` for more information on a specific monster',
+                value='or use `.monster vote {monster}` to cast your vote for said monster.',
+                inline=False
+            )
+            vote_embed.add_field(
+                name='You may vote for or show the following monsters:',
+                value=f"{', '.join(self.voter_registry.keys())}"
+            )
+            return await ctx.send(embed=vote_embed)
+        for monster in self.voter_registry.keys():
+            if ctx.author.id in self.voter_registry[monster]['votes']:
+                if name != monster:
+                    self.voter_registry[monster]['votes'].remove(ctx.author.id)
+                    break
+                else:
+                    vote_embed.add_field(
+                        name='Vote unsuccessful.',
+                        value='You already voted for this monster. '
+                              'If you want to change your vote, use another monster.',
+                        inline=False
+                    )
+                    await ctx.send(embed=vote_embed)
+                    await asleep(.5)
+                    return await ctx.invoke(self.monster_vote)
+        self.voter_registry[name]['votes'].append(ctx.author.id)
+        vote_embed.add_field(
+            name='Vote successful!',
+            value=f'You have successfully voted for {self.voter_registry[name]["full_name"]}!',
+            inline=False
+        )
+        vote_embed.set_thumbnail(url=self.voter_registry[name]['image'])
+        self.json_write()
+        return await ctx.send(embed=vote_embed)
+
+    @monster_group.command(
+        name='show'
+    )
+    async def monster_show(self, ctx: Context, name: str):
+        """
+        Shows the named monster. If one is not named, it sends the default voting embed instead.
+        :param ctx:
+        :param name:
+        :return:
+        """
+        m = self.voter_registry.get(name)
```

Comment:
A more robust search needs to be made here:

- Current search is case sensitive. e.g. `.ms show Dracula` doesn't match anything
- It's not intuitive or clear to the user how to show a specific monster until the blank vote embed is shown. Because you're matching on keys here, monster names don't align with the `full_name` being shown on the leaderboard. e.g. `.ms show Count Dracula` doesn't match anything.
- Might be worth also accepting an integer input to show the corresponding monster from the leaderboard